mmseg/models/utils/san_layers.py

```diff
@@ -0,0 +1,430 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Modified from https://github.com/MendelXu/SAN/blob/main/san/model/attn_helper.py  # noqa: E501
+# Copyright (c) 2023 MendelXu.
+# Licensed under the MIT License
+
+import warnings
+from typing import Optional
+
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+
+from mmseg.models.backbones.vit import TransformerEncoderLayer
+
+
+def cross_attn_with_self_bias(self, query, key, value, attn_mask=None):
+    """Implementation of cross attention layer which shares the embedding
+    weights with self-attention.
+
+    Args:
+        self: self-attention layer
+        query, key, value: map a query and a set of key-value pairs to
+        an output. See "Attention Is All You Need" for more details.
+        attn_mask: mask that prevents attention to certain positions.
+    """
+    return cross_attn_with_self_bias_func(
+        query,
```

Comment:
Can the two methods be combined into one? And then the naming of the `self` param is a bit misleading.